function/spring-splitter-function/src/main/java/org/springframework/cloud/fn/splitter/SplitterFunctionConfiguration.java

```diff
@@ -60,13 +61,13 @@ public Function<Message<?>, List<Message<?>>> splitterFunction(
 		AbstractMessageSplitter messageSplitter = expressionSplitter.or(() -> fileSplitter)
 			.or(() -> defaultSplitter)
 			.get();
+
 		messageSplitter.setApplySequence(splitterFunctionProperties.isApplySequence());
-		ThreadLocalFluxSinkMessageChannel outputChannel = new ThreadLocalFluxSinkMessageChannel();
+		FluxMessageChannel inputChannel = new FluxMessageChannel();
+		inputChannel.subscribe(messageSplitter);
+		FluxMessageChannel outputChannel = new FluxMessageChannel();
 		messageSplitter.setOutputChannel(outputChannel);
-		return (message) -> {
-			messageSplitter.handleMessage(message);
-			return outputChannel.publisherThreadLocal.get();
-		};
+		return (messageFlux) -> Flux.from(outputChannel).doOnRequest((l) -> inputChannel.subscribeTo(messageFlux));
```

Comment:
I find using the `__` var name for unused lambda parameters makes the code a tiny bit more readable. Wdyt?

Reply:
Sure thing!
It is more typing, though, plus it looks a bit weird: `(__) ->`.
According to our Checkstyle rules.
But I'll give it a shot! 😄